packages/quantic/force-app/main/default/lwc/quanticResultsPerPage/__tests__/quanticResultsPerPage.test.js

```diff
@@ -0,0 +1,360 @@
+        choicesDisplayed: customChoices.join(','),
+      });
+      await flushPromises();
+
+      const initializationError = element.shadowRoot.querySelector(
+        selectors.componentError
+      );
+      expect(initializationError).not.toBeNull();
+
+      expect(console.error).toHaveBeenCalledTimes(customChoices.length);
+      customChoices.forEach((choice, index) => {
+        expect(console.error).toHaveBeenNthCalledWith(
+          index + 1,
+          `The choice value "${choice}" from the "choicesDisplayed" option is not a number.`
+        );
+      });
+    });
+  });
+
+  describe('with a custom #initialChoice option', () => {
+    const customInitialChoiceIndex = 2;
+    const customInitialChoice = defaultChoices[customInitialChoiceIndex];
+
+    beforeEach(() => {
+      console.error = jest.fn();
+      resultsPerPageState = {
+        ...resultsPerPageState,
+        numberOfResults: customInitialChoice,
+      };
+    });
+
+    afterEach(() => {
+      cleanup();
+    });
+
+    it('should properly render with the initial choice selected', async () => {
+      const element = createTestComponent({
+        initialChoice: customInitialChoice,
+      });
+      await flushPromises();
+
+      expect(functionsMocks.buildResultsPerPage).toHaveBeenCalledTimes(1);
+      expect(functionsMocks.buildResultsPerPage).toHaveBeenCalledWith(
+        exampleEngine,
+        {
+          initialState: {
+            numberOfResults: customInitialChoice,
+          },
+        }
+      );
+
+      const resultsPerPageOptions = Array.from(
+        element.shadowRoot.querySelectorAll(selectors.resultsPerPageOption)
+      );
+      expect(resultsPerPageOptions).toHaveLength(defaultChoices.length);
+
+      resultsPerPageOptions.forEach((option, index) => {
+        expect(option.selected).toBe(index === customInitialChoiceIndex);
+      });
+    });
+    it('should log an error if the initial choice is not included in the choices displayed but not display it to users', async () => {
+      const customInvalidInitialChoice = 123;
+      const element = createTestComponent({
+        initialChoice: customInvalidInitialChoice,
+      });
+      await flushPromises();
+
+      expect(functionsMocks.buildResultsPerPage).toHaveBeenCalledTimes(1);
+      expect(functionsMocks.buildResultsPerPage).toHaveBeenCalledWith(
+        exampleEngine,
+        {
+          initialState: {
+            numberOfResults: defaultChoices[0],
+          },
+        }
+      );
+
+      const initializationError = element.shadowRoot.querySelector(
+        selectors.componentError
+      );
+      expect(initializationError).toBeNull();
+
+      expect(console.error).toHaveBeenCalledTimes(1);
+      expect(console.error).toHaveBeenCalledWith(
+        `The "initialChoice" option value "${customInvalidInitialChoice}" is not included in the "choicesDisplayed" option "${defaultChoices}". Defaulting to the first value of the choices.`
+      );
```

Comment:
I think it would also be useful here to check that the first option is selected in the HTML,
something like:
```
      resultsPerPageOptions.forEach((option, index) => {
        expect(option.selected).toBe(index === 0);
      });
```

Reply:
If the first option is selected depends entirely on our mock of Headless ResultsPerPage state, and not on internal state of the component.

I created a separate test to make sure the headless state was properly represented as selected.